pyhmmer/easel.pyx

```diff
@@ -5871,13 +5875,21 @@ cdef class DigitalSequenceBlock(SequenceBlock):
         for i in range(self._length):
             assert self._refs != NULL
             assert self._refs[i] != NULL
-            # get length of input and output sequences         
+            # get length of input and output sequences
             ntlen = self._refs[i].n
             if ntlen % 3 != 0:
                 raise ValueError(f"Incomplete sequence of length {ntlen!r} at index {i!r}")
             aalen = ntlen // 3
             # create new object
-            protein = DigitalSequence(genetic_code.amino_alphabet)
+            protein = DigitalSequence(
+                genetic_code.amino_alphabet,
+                name=self._refs[i].name,
+                description=self._refs[i].desc,
+                accession=self._refs[i].acc,
+                source=self._refs[i].source,
+                taxonomy_id=self._refs[i].tax_id
+            )
```

Comment:
Use `self._storage[i]` instead of `self._refs[i]` to use the attributes from Python objects rather than from the C objects directly.